client/network/src/on_demand_layer.rs

```diff
@@ -65,47 +65,47 @@ impl<Block: BlockT> FetchChecker<Block> for AlwaysBadChecker {
 		_remote_header: Option<Block::Header>,
 		_remote_proof: StorageProof,
 	) -> Result<Block::Header, ClientError> {
-		Err(ClientError::Msg("AlwaysBadChecker".into()))
+		Err(ClientError::AlwaysBadChecker)
```

Comment:
Is that really such a great idea to have a variant for this in the client error? Could we not use a local error type for this?

Reply:
Sure, we can.